### PR TITLE
Pre bench/Kafka test tidying

### PR DESCRIPTION
**Pull request by jarohen**
Have split out the parts of my Kafka PR that don't have anything to do with Kafka (so that these can be merged while I'm still working on the ingest performance)

* Tests that don't need to depend on the innards of Kafka now just depend on a started node
* Tests that have a tx in hand don't need to use sync-to-no-lag
* Various utilities/micro-refactorings/etc

### CODE REVIEW

File: crux-core/src/crux/db.clj [hunk 1]
```diff
@@ -23,6 +23,11 @@
   (read-index-meta [this k]))
 ;; end::Indexer[]
 
+(defn swap-index-meta [indexer k f & args]
```
Comment (jonpither) [inline crux-core/src/crux/db.clj]: Is this used anywhere?

File: crux-rdf/src/crux/rdf.clj [hunk 2]
```diff
@@ -208,14 +208,15 @@
...
+                 (when (zero? (long (mod entity-count *ntriples-log-size*)))
+                   (log/debug "submitted" entity-count))
                  (let [tx-ops (vec (for [entity entities]
                                      [:crux.tx/put entity]))]
-                   (db/submit-tx tx-log tx-ops))
-                 (+ n (count entities)))
-               0)))
+                   {:entity-count (+ entity-count (count entities))
+                    :last-tx (db/submit-tx tx-log tx-ops)}))
+
```
Comment (jonpither) [inline crux-rdf/src/crux/rdf.clj:218]: blank line needed?

File: crux-test/test/crux/api_test.clj [hunk 3]
```diff
@@ -81,9 +81,6 @@
   (t/testing "empty db"
     (t/is (.db *api*)))
 
-  (t/testing "syncing empty db"
-    (t/is (nil? (.sync *api* (Duration/ofSeconds 10)))))
```
Comment (jonpither) [inline crux-test/test/crux/api_test.clj]: Maybe you could note why this isn't useful - we did have an issue with performing operations on an empty DB in the past, so it's reassuring to see this test

File: crux-test/test/crux/fixtures/lubm.clj [hunk 4]
```diff
@@ -10,8 +10,12 @@
   (let [tx-ops (->> (concat (rdf/->tx-ops (rdf/ntriples "lubm/univ-bench.ntriples"))
                             (rdf/->tx-ops (rdf/ntriples lubm-triples-resource-8k)))
                     (rdf/->default-language)
-                    vec)]
-    (doseq [tx-ops (partition-all 1000 tx-ops)]
-      (api/submit-tx *api* (vec tx-ops)))
-    (api/sync *api* nil)
+                    vec)
+
```
Comment (jonpither) [inline crux-test/test/crux/fixtures/lubm.clj]: empty blank line

File: crux-test/test/crux/kafka_test.clj [hunk 5]
```diff
@@ -18,12 +18,16 @@
...
            java.util.List
            org.apache.kafka.clients.producer.ProducerRecord
+           org.apache.kafka.clients.consumer.ConsumerRecord
            org.apache.kafka.common.TopicPartition
            java.io.Closeable))
 
 (t/use-fixtures :once fk/with-embedded-kafka-cluster)
 (t/use-fixtures :each fk/with-kafka-client fkv/with-memdb fkv/with-kv-store)
 
+(defn consumer-record->value [^ConsumerRecord record]
```
Comment (jonpither) [inline crux-test/test/crux/kafka_test.clj]: Could be private

File: crux-test/test/crux/ts_devices_test.clj [hunk 6]
```diff
@@ -4,97 +4,94 @@
...
 (defn with-ts-devices-data [f]
   (if run-ts-devices-tests?
-    (let [submit-future (future (submit-ts-devices-data *api*))]
-      (api/sync *api* (java.time.Duration/ofMinutes 20))
-      (t/is (= 1001000 @submit-future))
+    (let [{:keys [op-count last-tx]} (submit-ts-devices-data *api*)]
+      (assert (= op-count 1001000) (str "actual op-count: " op-count))
+      (api/sync *api* (:crux.tx/tx-time last-tx) (java.time.Duration/ofMinutes 20))
       (f))
+
```
Comment (jonpither) [inline crux-test/test/crux/ts_devices_test.clj:86]: empty blank line

**Review by jonpither — APPROVED** [review]
Some very minor nitpicks and a query. I'd be happy to see this merged.

File: crux-rdf/src/crux/rdf.clj (hunk 2 above)
Comment (jarohen) [reply crux-rdf/src/crux/rdf.clj:218]: I tend to, gives more of a visual clue that the `{:entity-count 0}` isn't part of the function?

File: crux-test/test/crux/api_test.clj (hunk 3 above)
Comment (jarohen) [reply crux-test/test/crux/api_test.clj]: think it was part of removing the sync-to-no-lag commit (I cherry picked a few), will re-instate.

File: crux-test/test/crux/fixtures/lubm.clj (hunk 4 above)
Comment (jarohen) [reply crux-test/test/crux/fixtures/lubm.clj]: I like whitespace, apparently :smile:

File: crux-core/src/crux/db.clj (hunk 1 above)
Comment (jarohen) [reply crux-core/src/crux/db.clj]: is on my other branch :smile: removed

File: crux-test/test/crux/fixtures/lubm.clj (hunk 4 above)
Comment (jarohen) [reply crux-test/test/crux/fixtures/lubm.clj]: refactored

**Pull request merged**